### ISSUE TITLE
@Provides in different file

### ISSUE THREAD

**Issue by kozubikmichal** [thiagobustamante/typescript-ioc#11]
Hi,

I have issue using your IoC when the abstract class and providing class are in different files.

This is working just fine:

```typescript
import { Inject, Provides } from "typescript-ioc"

abstract class IFoo {
	abstract bar()
}

@Provides(IFoo)
class Foo implements IFoo {
	bar() {
		console.log("Foo bar");
	}
}

class Worker {
	@Inject
	private foo: IFoo

	work() {
		this.foo.bar()
	}
}

new Worker().work()
```

However when I split it into the 3 files I get the "**TypeError: this.foo.bar is not a function**" error:

index.ts:
```typescript
import { Inject } from "typescript-ioc"
import IFoo from "./IFoo";
import Foo from "./Foo";

class Worker {
	@Inject
	private foo: IFoo

	work() {
		this.foo.bar()
	}
}

new Worker().work()
```

IFoo.ts:
```typescript IFoo.ts
export default abstract class IFoo {
	abstract bar()
}
```

Foo.ts:
```typescript 
import { Provides } from "typescript-ioc"
import IFoo from "./IFoo"

@Provides(IFoo)
export default class Foo implements IFoo {
	bar() {
		console.log("Foo bar");
	}
}
```

**Comment by thiagobustamante** [comment]
Hi,

Thank for reporting. I'm out of home this week, on vacation with the kids. But I'll be back on the 25th and I promise to look at it as soon as I get home.

**Comment by thiagobustamante** [comment]
I added a test case that reproduces the problem. I will take a look at it now

**Comment by kozubikmichal** [comment]
Any progress?

**Comment by thiagobustamante** [comment]
The problem is that typescript-ioc does not scan any folder looking for classes. Your classes must be previously imported.

```typescript
require('./Foo'); // or import { Foo } from "./Foo";

new Worker().work()
```
The above code would make it work.

So, I created another helper method that can help to include sources in the container context for those code that are not already referenced. (You only need to include things that are not already imported).

```typescript
import { ContainerConfig } from "typescript-ioc/container-config";

ContainerConfig.addSource('lib/*'); // You can use glob patterns here
// or
ContainerConfig.addSource('controllers/*', 'baseFolder');
// or 
ContainerConfig.addSource(['**/*', '!foo.js'], 'baseFolder');
```

You need to configure those sources only once, but before you try to use the objects that depends on these files

**Comment by kozubikmichal** [comment]
Unfortunately I'm not able to get this work. I created a new project with the most actual version of typescript-ioc package and when I try to import ContainerConfig as described above, I'm getting ts error **Cannot find module 'typescript-ioc/container-config'**. I looked at the node_modules folder and the file is present but for some reason typescript cannot find it

![image](https://user-images.githubusercontent.com/26189327/30151900-a2d3ca2c-93b1-11e7-8489-87a5589efa19.png)

**Comment by thiagobustamante** [comment]
Sorry, I found an error in publishing the package. Please try again with version 1.0.2

**Comment by kozubikmichal** [comment]
Still no success. My folder structure:

![image](https://user-images.githubusercontent.com/26189327/30261622-d12a6f7e-96cd-11e7-9ea7-edf1effcce50.png)

Foo.ts:

```
import { Provides } from "typescript-ioc"

import IFoo from "./IFoo";

@Provides(IFoo)
export default class Foo extends IFoo {
	bar() {
		console.log("bar called");
	}
}
```

IFoo.ts:
```
export default abstract class IFoo {
	abstract bar();
}
```

index.ts:
```
import { Inject, Container, Config } from "typescript-ioc";

import { ContainerConfig } from "typescript-ioc/container-config";

import IFoo from "./IFoo";

ContainerConfig.addSource(["**/*", "!index.ts"], 'src');

class Worker {
	@Inject
	private foo: IFoo;

	work() {
		this.foo.bar();
	}
}


new Worker().work();
```

tsconfig.json: 
```
{
  "compilerOptions": {
    "outDir": "./dist/",
    "sourceMap": true,
    "noImplicitAny": false,
    "module": "commonjs",
    "target": "es3",
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "lib": [
      "dom", "es6"
    ]
  },
  "include": [
    "./src/**/*"
  ]
}
```

when I run **tsc && node dist/index.js** I'm getting this error:

![image](https://user-images.githubusercontent.com/26189327/30261740-4d00c1d4-96ce-11e7-968a-c9e53f4a016c.png)

**Comment by thiagobustamante** [comment]
I did not test this library with ```"target": "es3"```. Can you make a test compiling it to ES5?

**Comment by kozubikmichal** [comment]
still the same issue with both ES5 and ES6
